stdlib/public/core/Misc.swift

```diff
@@ -132,3 +132,12 @@ public func _getTypeByMangledNameInContext(
   genericContext: UnsafeRawPointer?,
   genericArguments: UnsafeRawPointer?)
   -> Any.Type?
+
+@_silgen_name("swift_getElfReflectionInfo")
```

Comment:
Is there a way to conditionally define these on ELF targets only?

Reply:
Hmmm, it seems that I can only specify 'os', 'arch', or 'swift'. I can conditionally define on "not mac and debug"

Reply:
Sounds good, I think that's exactly what we want. Note that some care is then needed to enable the reflection tests. We'd need some kind of "REQUIRES: basic_reflection_support" line, where the lit.cfg makes the "basic_reflection_support" feature available if "Mac || debug".

Reply:
This is incorrect. Below, you are using NDEBUG in the c++ code. This should be based off of something that is controlled by NDEBUG. Also, not Mac is not sufficient since this would be true on Windows as well. @compnerd

Reply:
As @gottesmm rightly pointed out - this doesn't seem correct.  Windows is not ELF, it is COFF, this seems like it should be further generalized and just be using the runtime information already present.